airbyte-webapp/src/components/connection/UpdateConnectionDataResidency/UpdateConnectionDataResidency.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { ControlLabels } from "components/LabeledControl";
+import { Card } from "components/ui/Card";
+import { DropDown } from "components/ui/DropDown";
+import { Spinner } from "components/ui/Spinner";
+
+import { Geography } from "core/request/AirbyteClient";
+import { useConnectionEditService } from "hooks/services/ConnectionEdit/ConnectionEditService";
+import { useNotificationService } from "hooks/services/Notification";
+import { useAvailableGeographies } from "packages/cloud/services/geographies/GeographiesService";
+import { links } from "utils/links";
+
+import styles from "./UpdateConnectionDataResidency.module.scss";
+
+export const UpdateConnectionDataResidency: React.FC = () => {
+  const { connection, updateConnection, connectionUpdating } = useConnectionEditService();
+  const { registerNotification } = useNotificationService();
+  const { formatMessage } = useIntl();
+  const [selectedValue, setSelectedValue] = useState<Geography>();
+
+  const { geographies } = useAvailableGeographies();
+
+  const handleSubmit = async ({ value }: { value: Geography }) => {
+    try {
+      setSelectedValue(value);
+      await updateConnection({
+        connectionId: connection.connectionId,
+        geography: value,
+      });
+    } catch (e) {
+      registerNotification({
+        id: "connection.geographyUpdateError",
+        title: formatMessage({ id: "connection.geographyUpdateError" }),
+        isError: true,
+      });
+    }
+    setSelectedValue(undefined);
+  };
+
+  return (
+    <Card withPadding>
+      <div className={styles.wrapper}>
+        <div>
+          <ControlLabels
+            nextLine
+            label={<FormattedMessage id="connection.geographyTitle" />}
+            message={
+              <FormattedMessage
+                id="connection.geographyDescription"
+                values={{
+                  lnk: (node: React.ReactNode) => (
+                    <a href={links.cloudAllowlistIPsLink} target="_blank" rel="noreferrer">
+                      {node}
+                    </a>
+                  ),
+                }}
+              />
+            }
+          />
+        </div>
+        <div className={styles.dropdownWrapper}>
+          <div className={styles.spinner}>{connectionUpdating && <Spinner small />}</div>
```

Comment:
Not for this PR, but I feel we will have this use-case more often and might want to have a loading state in the dropdown component itself? Maybe worth creating an Asana issue for Nico to look into that.

Reply:
Good point, I brought it up with Nico but we didn't find a good solution yet. Created an Asana issue here: https://app.asana.com/0/1202192547291053/1203300180185946/f

I think we also need to consider loading / success / error states for these interactions, since we don't have a standard pattern yet.